ApiReview/API.Polly.Core/NoDocs/Polly.Timeout/TimeoutStrategyOptions.cs

```diff
@@ -0,0 +1,17 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Polly.Timeout;
+
+public class TimeoutStrategyOptions : ResilienceStrategyOptions
+{
+    public sealed override string StrategyType { get; }
+    [Timeout]
+    public TimeSpan Timeout { get; set; }
+    public Func<TimeoutGeneratorArguments, ValueTask<TimeSpan>>? TimeoutGenerator { get; set; }
```

Comment:
@andrey-noskov So Andrey, in this PR's API there are a bunch of delegates passed in through the option objects. This basically implies that these delegate implementations cannot use any state coming from DI.

Is that OK? Is there a workaround if you really need a delegate that has access to DI-supplied types?